userservice/src/main/java/com/infodation/userservice/services/UserServiceImpl.java

```diff
@@ -21,7 +26,65 @@ public class UserServiceImpl implements IUserService {
     public UserServiceImpl(UserRepository userRepository) {
         this.userRepository = userRepository;
     }
+    @Async  // Annotation for asynchronous execution
+    public CompletableFuture<Void> importUsersFromCsvAsync(MultipartFile file) throws IOException {
+        List<User> usersToSave = new ArrayList<>();
+        // Retrieve all existing userIds from the database (for comparison)
+        List<String> allUserIdsInDb = userRepository.findAllUserIds();
+        Set<String> userIdsInDbSet = new HashSet<>(allUserIdsInDb);
 
+        // Read the CSV file
+        try (CSVReader reader = new CSVReader(new BufferedReader(new InputStreamReader(file.getInputStream())))) {
+            String[] nextLine;
+            boolean isHeader = true;
+
+            while ((nextLine = reader.readNext()) != null) {
+                if (isHeader) {  // Skip the header row
+                    isHeader = false;
+                    continue;
+                }
+
+                try {
+                    String userId = nextLine[1];  // column 1 is userId
+
+                    // Check if userId already exists in the database
+                    if (userIdsInDbSet.contains(userId)) {
+                        System.out.println("User with ID " + userId + " already exists, skipping.");
+                        continue; // Skip if userId already exists in the database
+                    }
+                    // Map the CSV data to a User object
+                    User user = new User();
+                    user.setUserId(userId); // Column 1 is userId
+                    user.setFirstName(nextLine[2]); // Column 2 is firstName
+                    user.setLastName(nextLine[3]); // Column 3 is lastName
+                    user.setEmail(nextLine[5]); // Column 5 is email
+                    user.setSex(Sex.valueOf(nextLine[4].toUpperCase())); // Column 4 is MALE/FEMALE
+                    user.setCreatedAt(LocalDateTime.now());
+                    user.setUpdatedAt(LocalDateTime.now());
+                    // Add user to the list if it doesn't exist in the database
+                    usersToSave.add(user);
+                    // Save when the list reaches 5,000 users
+                    if (usersToSave.size() == 5000) {
```

Comment:
It is good to define this as a constant variable or define as a configuration of the application.
But I prefer setting this as a configuration (in `application.properties` file) and you can you annotation `@Value` of Springboot framework to get in your Service class.
Example: https://www.baeldung.com/spring-value-annotation